22/7/11(월) PR#1

wygbsr opened the pull request.

findstar committed:
객체지향 생활체조 9원칙 추가

wygbsr committed:
최초 커밋입니다.

findstar commented on src/main/java/org/sscc/study/SetCollections.java:
```diff
@@ -0,0 +1,6 @@
+package org.sscc.study;
+import java.util.*;
+
+public class SetCollections {
```
학습테스트에서는 굳이 비어있는 클래스를 만들 필요는 없습니다 :)

findstar commented on src/test/java/org/sscc/study/SetTest.java:
```diff
@@ -0,0 +1,51 @@
+
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.params.*;
+import org.junit.jupiter.params.provider.*;
```
import를 할 때 와일드카드를 사용하셨는데 가급적이면 명시적인 클래스 import를 선언하는 것이 좋습니다.

findstar commented on src/test/java/org/sscc/study/SetTest.java:
```diff
@@ -0,0 +1,51 @@
+        numbers.add(3);
+    }
+
+    @Test
```
테스트가 어떤 기능을 테스트 하는지 명확하게 알기가 어렵습니다.
<img width="364" alt="image" src="https://user-images.githubusercontent.com/1266944/178421441-53193495-1dc5-4bab-b42c-a4745ac74d53.png">

StringClassTest에서와 같이 
DisplayName 을 추가 해보시는건 어떨까요?

findstar commented on src/test/java/org/sscc/study/SetTest.java:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    @ParameterizedTest
+    @CsvSource({
```
요렇게 알아보기 쉽게 정리된 것 좋습니다 👍

findstar commented on src/test/java/org/sscc/study/SetTest.java:
```diff
@@ -0,0 +1,51 @@
+import java.util.*;
+
+import static org.assertj.core.api.Assertions.*;
+
```
import를 static 과 그렇지 않은 경우로 구분한 경우가 아니라면 
그룹 별로 띄워놓지 않아도 됩니다.

IntelliJ 에서 code reformat 기능을 활용해보세요.

wygbsr committed:
Refactor: Reflect feedbacks and rename test methods

Remove empty classes
Import classes explicitly
 -Group import statements into non-static and static
Use @DisplayName to describe test methods
 -Rename test methods in (should_ExpectedBehavior_When_StateUnderTest) format for better readability
 -(MethodName_StateUnderTest_ExpectedBehavior) format is avoided in case of method refactoring

findstar reviewed (approved):
확인했습니다. 수고하셨습니다.
